### ISSUE TITLE
[Enhancement] Add unit tests for `pkg/etcdutil`

### ISSUE THREAD

**Issue by ishan16696** [gardener/etcd-backup-restore#800]
**Enhancement (What you would like to be added):**
Right now, there is no unit tests present for `pkg/etcdutil`. This issue is to add unit tests to `pkg/etcdutil`.

**Motivation (Why is this needed?):**

**Approach/Hint to the implement solution (optional):**

**Comment by ishan16696** [comment]
Added the unit tests for  etcdutil `func TakeAndSaveFullSnapshot()` in this PR: https://github.com/gardener/etcd-backup-restore/pull/779

**Comment by ishan16696** [comment]
I found out that unit tests for most of functions present inside `pkg/etcdutil` are already present but they are in `pkg/defragmentor`.
https://github.com/gardener/etcd-backup-restore/blob/4703c603a2e17a29416280aea637040d877b1a89/pkg/defragmentor/defrag_test.go#L52-L199

@shreyas-s-rao any opinion on this, should I move them to `pkg/etcdutil` ?

**Comment by shreyas-s-rao** [comment]
Yes, moving them would be good, thanks! Because these tests don't really belong in `defragmentor` package.

**Comment by ishan16696** [comment]
I have opened the PR https://github.com/gardener/etcd-backup-restore/pull/837 to move the unit tests.
